mlog/log.go

```diff
@@ -26,6 +27,15 @@ const (
 	LevelError = "error"
 )
 
+var (
+	// disableZap is set when Zap should be disabled and Logr used instead.
+	// This is needed for unit testing as Zap has no shutdown capabilities
+	// and holds file handles until process exit. Currently unit test create
+	// many server instances, and thus many Zap log files.
+	// This flag will be removed when Zap is permanently replaced.
+	disableZap bool
```

Comment:
I understand this is temporary. But having an unguarded global boolean variable can be racy.

Also, is there a ticket for zap removal? We should add that here.

Reply:
It is racy.  Not likely to be an issue now, but I'll guard it in case it gets accessed differently later.

Reply:
I've made the flag atomic.  Thanks.  https://github.com/mattermost/mattermost-server/pull/15398/commits/3fc53b99414463cd7d2a7b5e92c0fc3acf5ad75a

Reply:
Also added ticket for Zap removal:  https://mattermost.atlassian.net/browse/MM-28405